### PR TITLE
Questionnaire UI

### PR DESCRIPTION
**Pull request by aeyates**
Display historical responses and allow for submission of new responses.
new-response
historical-responses

### CODE REVIEW

**Review by wsams — APPROVED** [review]
This looks good @aeyates . These changes are easy to understand. I took a look at the whole file. It is getting pretty large. I see a couple areas that could benefit from a component, but not sure if it's worth the extra time for a demo.

**Comment by aeyates** [comment]
Thanks @wsams . Yes, I actually did try to break one piece out into a component but could never get it to work that way. I hope to do a pass once this dashboard is closer to finalized. There are still a number of changes coming.

File: src/main/resources/static/assets/js/sdh.js [hunk 1]
```diff
@@ -275,8 +305,46 @@ class ModalEditRow {
...
+							}
+						});
+					}
+				});
+		    	
 		    }
 		},
+		watch: {
+			selectedResponseDate: function(newDate, oldDate) {
+				// This triggers the UI to update the selectedResponse
```
Comment (heathharrelson) [inline src/main/resources/static/assets/js/sdh.js:345]: It seems kind of weird to have a watcher that doesn't perform any actions of its own. Could this be replaced by a change handler on the field itself?

Comment (aeyates) [reply src/main/resources/static/assets/js/sdh.js:345]: This was one of the blockers I ran into. The change handler was always capturing the state previous to the change. I'm guessing this means there is some interaction in computed properties that is not quite right, but after many attempts I landed on using watch. I'd definitely like to revisit this one.

**Review by heathharrelson — APPROVED** [review]
The code for the component is definitely getting unwieldy. If you want to pair on extracting additional components or adding a webpack build, I can always make time to help.

File: src/main/resources/static/assets/js/sdh.js (hunk 1 above)
Comment (heathharrelson) [reply src/main/resources/static/assets/js/sdh.js:345]: I've definitely had some frustrations related to computed properties not updating as expected.

I'm not super familiar with the Bootstrap components you're using, but with a plain select list, the change handler would give you an event object that you could use to get the new selection (via event.target.value), but I'm surprised the v-model isn't doing the right thing. Have you tried using [`$nextTick`](https://vuejs.org/v2/api/#vm-nextTickhttps://vuejs.org/v2/api/#vm-nextTick) to wait for the next turn of the run loop? Sometimes that helps.

Comment (aeyates) [reply src/main/resources/static/assets/js/sdh.js:345]: I stumbled on $nextTick later in relation to something else and wondered if that would have worked in this case but didn't try it.

**Pull request merged**